deploy/azureDeployFunction.js

```diff
@@ -18,6 +18,7 @@ class AzureDeployFunction {
 
     this.hooks = {
       'deploy:function:deploy': () => BbPromise.bind(this)
+        .then(this.provider.initialise(this.serverless,this.options))
```

Comment:
We need to do the initialise as late in the process as possible to ensure that the variable substitution has occurred. 

This provider seems to work differently, other providers appear to pass in all of the required parameters, rather than keeping them stored inside the provider like this one.

If there is a better place to put this I can change it - but this is as late in the process as I can find that ensures that the framework has configured everything we need.